compiler/src/dotty/tools/dotc/transform/TypeTestsCasts.scala

```diff
@@ -330,13 +330,17 @@ object TypeTestsCasts {
             expr.isInstance(testType).withSpan(tree.span)
           case OrType(tp1, tp2) =>
             evalOnce(expr) { e =>
-              transformTypeTest(e, tp1, flagUnrelated = false)
-                .or(transformTypeTest(e, tp2, flagUnrelated = false))
+              lazy val tp1Tree = transformTypeTest(e, tp1, flagUnrelated = false)
+              lazy val tp2Tree = transformTypeTest(e, tp2, flagUnrelated = false)
```

Comment:
Considering the case `String | (Null | Null)`, is it easier to call `stripNull` which can remove all `Null` from a union type?